pgserver/subscription_handler.go

```diff
@@ -0,0 +1,216 @@
+package pgserver
+
+import (
+	"context"
+	stdsql "database/sql"
+	"fmt"
+	"github.com/apecloud/myduckserver/adapter"
+	"github.com/apecloud/myduckserver/pgserver/logrepl"
+	"github.com/jackc/pglogrepl"
+	"regexp"
+)
+
+// This file implements the logic for handling CREATE SUBSCRIPTION SQL statements.
+// Example usage of CREATE SUBSCRIPTION SQL:
+//
+// CREATE SUBSCRIPTION mysub
+// CONNECTION 'dbname= host=127.0.0.1 port=15432 user=postgres password=root'
+// PUBLICATION mypub;
+//
+// The statement creates a subscription named 'mysub' that connects to a PostgreSQL
+// database and subscribes to changes published under the 'mypub' publication.
+
+type SubscriptionConfig struct {
+	SubscriptionName string
+	PublicationName  string
+	DBName           string
+	Host             string
+	Port             string
+	User             string
+	Password         string
+	LSN              pglogrepl.LSN
+}
+
+var lsnQueryIndex = 5
+
+var subscriptionRegex = regexp.MustCompile(`CREATE SUBSCRIPTION\s+(\w+)\s+CONNECTION\s+'([^']+)'\s+PUBLICATION\s+(\w+);`)
```

Comment:
I think it would be better to create a regex object which can match the string in case-incensitive way.

Reply:
fixed